From 930639722342d82d7ffe7dbc14c0b93afd8c322d Mon Sep 17 00:00:00 2001
From: Li Wencheng <liwencheng@phytium.com.cn>
Date: Mon, 29 Apr 2024 18:23:45 +0800
Subject: [PATCH 24/67] PHYTIUM: net/phytmac: Bugfix MAC address change when
 rmmod and insmod module

The eth_hw_addr_random function cause systemd-udevd to generate
MAC address, which will modify the default MAC address.

Mainline: NA
Signed-off-by: Li Wencheng <liwencheng@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
Change-Id: Ie9fa00c3a4c9310104fd888526571df95f15c664

Link: https://gitee.com/phytium_embedded/phytium-linux-kernel/commit/6302b0ab85a13bec77c34afd9e0a015ac3b0ceda
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/net/ethernet/phytium/phytmac.h      | 2 +-
 drivers/net/ethernet/phytium/phytmac_main.c | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/phytium/phytmac.h b/drivers/net/ethernet/phytium/phytmac.h
index 6eb1d7d1b829..cb9359c7bd4d 100644
--- a/drivers/net/ethernet/phytium/phytmac.h
+++ b/drivers/net/ethernet/phytium/phytmac.h
@@ -13,7 +13,7 @@

 #define PHYTMAC_DRV_NAME		"phytium-mac"
 #define PHYTMAC_DRV_DESC		"PHYTIUM Ethernet Driver"
-#define PHYTMAC_DRIVER_VERSION		"1.0.1"
+#define PHYTMAC_DRIVER_VERSION		"1.0.2"
 #define PHYTMAC_DEFAULT_MSG_ENABLE	  \
 		(NETIF_MSG_DRV		| \
 		NETIF_MSG_PROBE	| \
diff --git a/drivers/net/ethernet/phytium/phytmac_main.c b/drivers/net/ethernet/phytium/phytmac_main.c
index d932d011b261..732358ea40af 100644
--- a/drivers/net/ethernet/phytium/phytmac_main.c
+++ b/drivers/net/ethernet/phytium/phytmac_main.c
@@ -2208,7 +2208,6 @@ static int phytmac_init(struct phytmac *pdata)

 	ndev->netdev_ops = &phytmac_netdev_ops;
 	phytmac_set_ethtool_ops(ndev);
-	eth_hw_addr_random(pdata->ndev);

 	if (ndev->hw_features & NETIF_F_NTUPLE) {
 		INIT_LIST_HEAD(&pdata->rx_fs_list.list);
--
2.50.1
